### PR TITLE
Expose {ReadOnly}Span from System.Runtime

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/22353
cc: @weshaggard, @jkotas, @KrzysztofCwalina, @ahsonkhan

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet/dnceng, some Linux runs have been taking a very long time today.  Any idea why? e.g.
https://ci3.dot.net/job/dotnet_corefx/job/master/job/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/1548/

**Comment by maririos** [comment]
FYI @smile21prc

**Comment by MattGal** [comment]
@stephentoub  it's because Azure has removed Ubuntu 16.10 images so we only have one VM for every run.  If we can get some clear messaging on whether we need to test on the EOL'ed operating system we can disable it in a wide variety of ways.

File: src/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -1957,6 +1989,41 @@ public partial struct Single : System.IComparable, System.IComparable<float>, Sy
...
         public static bool TryParse(string s, out float result) { throw null; }
     }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public partial struct Span<T>
+    {
+        public static Span<T> Empty { get { throw null; } }
+        public Span(T[] array) { throw null; }
+        public Span(T[] array, int start) { throw null; }
+        public Span(T[] array, int start, int length) { throw null; }
+        [CLSCompliant(false)]
```
Comment (ahsonkhan) [inline src/System.Runtime/ref/System.Runtime.cs]: Can you please explain the difference between [CLSCompliant(false)] and [System.CLSCompliantAttribute(false)] that I see elsewhere (such as https://github.com/dotnet/corefx/blob/master/src/System.Threading.ThreadPool/ref/System.Threading.ThreadPool.cs#L34)?

Comment (stephentoub) [inline src/System.Runtime/ref/System.Runtime.cs]: The resulting IL is identical: the C# language allows the "Attribute" suffix to be omitted on attributes, and we're in the System namespace here so it's also superfluous.  I didn't see that the style throughout the rest of the file was to fully spell everything out, but to match that I probably should have done so.  I'll merge this and then fix it in the other System.Runtime changes I have pending, so as not to need to restart CI for this.

**Comment by stephentoub** [comment]
> it's because Azure has removed Ubuntu 16.10 images so we only have one VM for every run. If we can get some clear messaging on whether we need to test on the EOL'ed operating system we can disable it in a wide variety of ways.

We need to do something.  This leg is taking more than three hours in multiple PRs.  Either we should add more resources, or we should remove it / replace it in the leg with whatever OS we actually care about. @danmosemsft? @karelz?

**Comment by weshaggard** [comment]
>We need to do something. This leg is taking more than three hours in multiple PRs. Either we should add more resources, or we should remove it / replace it in the leg with whatever OS we actually care about. @danmosemsft? @karelz?

I'm going to remove 16.10 from the CI test queue list https://github.com/dotnet/corefx/pull/22568.

File: src/System.Runtime/ref/System.Runtime.cs (hunk 1 above)
Comment (weshaggard) [inline src/System.Runtime/ref/System.Runtime.cs]: The difference generally comes from whether someone generated this with our tools (the longer fully qualified version) vs someone manually authoring it which is usually the shortened version.

**Comment by stephentoub** [comment]
I'm going to merge this and several of the other PRs stuck on the Linux leg just because of Ubuntu 16.10.

**Pull request merged**